### PR TITLE
[ASC Track2 #5] migrate azure-toolkit-springcloud-lib to track2 API

### PR DESCRIPTION
**Pull request by wangmingliang-ms**

### CODE REVIEW

**Comment by andxu** [comment]
This PR is too huge, please split it smaller.

File: azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/AzureSpringCloud.java [hunk 1]
```diff
@@ -50,38 +53,56 @@ private AzureSpringCloud(@Nonnull final List<Subscription> subscriptions) { // f
...
+
+    @Cacheable(cacheName = "SpringServices", key = "$subscriptionId")
+    protected SpringServices getClient(final String subscriptionId) {
+        return getAzureResourceManager(subscriptionId).springServices();
     }
 
-    @Cacheable(cacheName = "AppPlatformManager", key = "$subscriptionId")
-    protected AppPlatformManager getClient(final String subscriptionId) {
+    @Cacheable(cacheName = "AzureResourceManager", key = "$subscriptionId")
+    private AzureResourceManager getAzureResourceManager(String subscriptionId) {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/AzureSpringCloud.java:88]: we shall replace AzureResourceManager with  dedicated manager

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java [hunk 2]
```diff
@@ -131,335 +63,47 @@
...
-        // init proxy manager
-        ProxyUtils.initProxy(Optional.ofNullable(this.session).map(MavenSession::getRequest).orElse(null));
-        // Init telemetries
-        initTelemetry();
-        telemetries.put(PROXY, String.valueOf(ProxyManager.getInstance().getProxy() != null));
-        Azure.az().config().setLogLevel(HttpLogDetailLevel.NONE.name());
-        Azure.az().config().setUserAgent(getUserAgent());
-        trackMojoExecution(MojoStatus.Start);
+    protected void beforeMojoExecution() {
+        super.beforeMojoExecution();
```
Comment (Flanker32) [inline azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java:72]: Whether we do authentication in each goal of spring plugin? Besides, do we need to do some static validation first before authentication.

File: azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/SpringCloudApp.java [hunk 3]
```diff
@@ -6,228 +6,254 @@
...
 import java.util.stream.Collectors;
 
 public class SpringCloudApp implements IAzureEntityManager<SpringCloudAppEntity> {
     private static final String UPDATE_APP_WARNING = "It may take some moments for the configuration to be applied at server side!";
 
     @Getter
-    private final SpringCloudCluster cluster;
-    private final SpringCloudAppManager appManager;
+    @Nonnull
+    final SpringCloudCluster cluster;
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/SpringCloudApp.java:37]: make it private? As we have getter for it

Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/SpringCloudApp.java:37]: I will remove the getter

File: azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/AzureSpringCloud.java (hunk 1 above)
Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/AzureSpringCloud.java:88]: yes, I will fix it.

**Pull request merged**